Ye grand old Walls Import PR

jedwards1211 opened the pull request:
Finally it's done!

Compiles successfully on my MBP and on virtualized Windows 10 (using DLLs), so everything should be good to go :)

vpicaver commented:
Sweet! I'll check this out tonight :D

On Tue, Dec 15, 2015 at 4:10 PM, Andy Edwards <notifications@github.com>
wrote:

> Finally it's done!
>
> Compiles successfully on my MBP and on virtualized Windows 10 (using
> DLLs), so everything should be good to go :)
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/Cavewhere/cavewhere/pull/94
> Commit Summary
>
>    - fix unused param compile error
>    - bump dewalls version
>    - fix Walls LRUD import and import surveyors from line 2
>    - first stab at wpj import
>    - refactor so that cwImportSurvexDialog will work with a
>    cwWallsImporter too
>    - use cwImportSurvexDialog with cwWallsImporter, woohoo\!
>    - put detached LRUD support back in
>    - apply relevant #units options from .wpj to each .srv file
>    - updte dewalls
>    - merge upstream/master
>    - big code cleanup
>    - update dewalls
>    - update dewalls
>    - update dewalls
>    - ignore survey nodes without a name and handle absolute node paths
>    correctly
>    - fix walls INCS corrections
>    - use Walls-specific labels in walls import dialog
>    - use trip titles from .wpj
>    - communicate with WallsParser using signals/slots
>    - fix optional station check
>    - add Walls calibration import test...currently failing because of
>    non-thread-safe Length singleton
>    - refactored dewalls units
>    - update dewalls
>    - add documentation and rename a few things
>    - refactor #segment parsing
>    - more code cleanup
>    - preemptively fix a bug
>    - more code cleanup
>    - extract a base class from cwSurvexGlobalData and rename some things
>    - implement caves() for cwWallsImportData
>    - added back the ability to import multiple Walls .srv files manually
>    - more renaming and sauce
>    - create cwSurvexNodeData...
>    - fix survex import
>    - added warnings for station renaming, refs, and fix stations
>    - fix windows compilation issues
>    - merge dewalls stuff
>    - create import errors list in cwImportTreeDataDialog and hook up
>    Walls Import warnings to it
>    - update dewalls
>    - Merging origin:wpj-import
>    - remove unused members
>    - update dewalls
>    - Merge branch 'wpj-import' of github.com:jedwards1211/cavewhere into
>    wpj-import
>    - fix windows linker errors
>    - show Walls Errors in tab in walls import dialog
>
> File Changes
>
>    - *M* dewalls
>    <https://github.com/Cavewhere/cavewhere/pull/94/files#diff-0> (2)
>    - *M* qml/ExportImportButtons.qml
>    <https://github.com/Cavewhere/cavewhere/pull/94/files#diff-1> (25)
>    - *M* src/cavewhereLib.qbs
>    <https://github.com/Cavewhere/cavewhere/pull/94/files#diff-2> (4)
>    - *R* src/cwImportTreeDataDialog.cpp
>    <https://github.com/Cavewhere/cavewhere/pull/94/files#diff-3> (127)
>    - *R* src/cwImportTreeDataDialog.h
>    <https://github.com/Cavewhere/cavewhere/pull/94/files#diff-4> (27)
>    - *R* src/cwImportTreeDataDialog.ui
>    <https://github.com/Cavewhere/cavewhere/pull/94/files#diff-5> (16)
>    - *M* src/cwSurvexGlobalData.cpp
>    <https://github.com/Cavewhere/cavewhere/pull/94/files#diff-6> (95)
>    - *M* src/cwSurvexGlobalData.h
>    <https://github.com/Cavewhere/cavewhere/pull/94/files#diff-7> (36)
>    - *M* src/cwSurvexImporter.cpp
>    <https://github.com/Cavewhere/cavewhere/pull/94/files#diff-8> (39)
>    - *M* src/cwSurvexImporter.h
>    <https://github.com/Cavewhere/cavewhere/pull/94/files#diff-9> (36)
>    - *D* src/cwSurvexImporterModel.cpp
>    <https://github.com/Cavewhere/cavewhere/pull/94/files#diff-10> (326)
>    - *A* src/cwSurvexNodeData.cpp
>    <https://github.com/Cavewhere/cavewhere/pull/94/files#diff-11> (44)
>    - *A* src/cwSurvexNodeData.h
>    <https://github.com/Cavewhere/cavewhere/pull/94/files#diff-12> (55)
>    - *M* src/cwSurveyImportManager.cpp
>    <https://github.com/Cavewhere/cavewhere/pull/94/files#diff-13> (99)
>    - *M* src/cwSurveyImportManager.h
>    <https://github.com/Cavewhere/cavewhere/pull/94/files#diff-14> (10)
>    - *A* src/cwTreeDataImporter.cpp
>    <https://github.com/Cavewhere/cavewhere/pull/94/files#diff-15> (42)
>    - *A* src/cwTreeDataImporter.h
>    <https://github.com/Cavewhere/cavewhere/pull/94/files#diff-16> (39)
>    - *A* src/cwTreeDataImporterModel.cpp
>    <https://github.com/Cavewhere/cavewhere/pull/94/files#diff-17> (326)
>    - *R* src/cwTreeDataImporterModel.h
>    <https://github.com/Cavewhere/cavewhere/pull/94/files#diff-18> (30)
>    - *A* src/cwTreeImportData.cpp
>    <https://github.com/Cavewhere/cavewhere/pull/94/files#diff-19> (29)
>    - *A* src/cwTreeImportData.h
>    <https://github.com/Cavewhere/cavewhere/pull/94/files#diff-20> (43)
>    - *R* src/cwTreeImportDataNode.cpp
>    <https://github.com/Cavewhere/cavewhere/pull/94/files#diff-21> (88)
>    - *R* src/cwTreeImportDataNode.h
>    <https://github.com/Cavewhere/cavewhere/pull/94/files#diff-22> (99)
>    - *A* src/cwWallsImportData.cpp
>    <https://github.com/Cavewhere/cavewhere/pull/94/files#diff-23> (109)
>    - *A* src/cwWallsImportData.h
>    <https://github.com/Cavewhere/cavewhere/pull/94/files#diff-24> (17)
>    - *M* src/cwWallsImporter.cpp
>    <https://github.com/Cavewhere/cavewhere/pull/94/files#diff-25> (556)
>    - *M* src/cwWallsImporter.h
>    <https://github.com/Cavewhere/cavewhere/pull/94/files#diff-26> (125)
>    - *A* testcases/cwWallsImporterTest.cpp
>    <https://github.com/Cavewhere/cavewhere/pull/94/files#diff-27> (71)
>
> Patch Links:
>
>    - https://github.com/Cavewhere/cavewhere/pull/94.patch
>    - https://github.com/Cavewhere/cavewhere/pull/94.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Cavewhere/cavewhere/pull/94>.
>

vpicaver commented:
Currently, I have cavewhere-test compiling in all of the dewall-test files. I'm getting the following error in cavewhere-test:

error: I couldn't open :/test/Kaua North Maze.wpj

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cavewhere-test is a Catch v1.2.1 host application.
Run with -? for options

-------------------------------------------------------------------------------
WallsProjectParser parses example file
-------------------------------------------------------------------------------
/Users/vpicaver/Documents/Projects/cavewhere/dewalls/test/wallsprojectparsertests.cpp:11
...............................................................................

/Users/vpicaver/Documents/Projects/cavewhere/dewalls/test/wallsprojectparsertests.cpp:23: FAILED:
  REQUIRE( !projectRoot.isNull() )
with expansion:
  false

===============================================================================
test cases:   17 |   16 passed | 1 failed
assertions: 6849 | 6848 passed | 1 failed

vpicaver commented:
We probably need to compile in a qrc file somewhere?

jedwards1211 commented:
Have you done a submodule update?  Or maybe you need to go into dewalls folder and do `git pull origin master`?  All the necessary files and present in the dewalls commit referenced in the wpj-import branch, I just checked.

If it still doesn't work check go into the dewalls folder and do `git log` and check if the head entry is https://github.com/Cavewhere/dewalls/tree/98d43d60aac7002f6f0952a59eea4af1dff2359b/test.  If not let me know

vpicaver commented:
Currently on 98d43d6

On Tue, Dec 15, 2015 at 7:49 PM, Andy Edwards <notifications@github.com>
wrote:

> Have you done a submodule update? Or maybe you need to go into dewalls
> folder and do git pull origin master? All the necessary files and present
> in the dewalls commit referenced in the wpj-import branch, I just checked.
>
> If it still doesn't work check go into the dewalls folder and do git log
> and check if the head entry is
> https://github.com/Cavewhere/dewalls/tree/98d43d60aac7002f6f0952a59eea4af1dff2359b/test.
> If not let me know
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Cavewhere/cavewhere/pull/94#issuecomment-164947785>.
>

jedwards1211 commented:
Oh!  I wasn't thinking, you only got the error when running cavewhere-test instead of dewalls-test.  I added dewalls-test.qrc and it works now.

Also thanks for sharing the projects that wouldn't parse, they helped me catch some important subtleties.  There were two problems with undocumented Walls parsing behavior that I fixed and added testcases for (it allows < 4 LRUDs and allows one - for omit even though some parts only say to use two -- for omit).  

The Harris Cave data now imports successfully.

The other problem, affecting TOFTData, is I was confused about what Walls' TAPE option meant, but now I've figured it out...I'll try to fix that tomorrow.

jedwards1211 commented:
(which by the way, note that some stations in the Harris Cave data only have 2 or 3 of the LRUD measurements given).
I think it's really unwise for Walls to accept that -- there's a good chance the user made a typo.  I should add warnings for this case to the importer

vpicaver commented:
Cool! I glad the dataset help. Are you updating the testcases to cover the
bugs you found? Cavewhere will also show warnings one the missing data.

On Wed, Dec 16, 2015 at 12:44 AM, Andy Edwards <notifications@github.com>
wrote:

> (which by the way, note that some stations in the Harris Cave data only
> have 2 or 3 of the LRUD measurements given).
> I think it's really unwise for Walls to accept that -- there's a good
> chance the user made a typo. I should add warnings for this case to the
> importer
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Cavewhere/cavewhere/pull/94#issuecomment-165002757>.
>

jedwards1211 commented:
Yes, I had to update the testcases, because they failed when I updated it to be as lenient as walls :)

The Big Spring data now parses successfully, though it's not yet deriving the distance/inclination from the instrument height/target height/inch correctly.  The math for that is a bit tedious...I want to make sure I have my head wrapped around it before I implement it

vpicaver commented:
Sounds good. I'm probably going  to be implementing dive data here soon. So
I can have full support for the meta cave format

On Wednesday, December 16, 2015, Andy Edwards <notifications@github.com>
wrote:

> Yes, I had to update the testcases, because they failed when I updated it
> to be as lenient as walls :)
>
> The Big Spring data now parses successfully, though it's not yet deriving
> the distance/inclination from the instrument height/target height/inch
> correctly. The math for that is a bit tedious...I want to make sure I have
> my head wrapped around it before I implement it
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Cavewhere/cavewhere/pull/94#issuecomment-165224380>.
>


-- 
Phi|ip

Sent from Gmail Mobile

vpicaver commented:
Just let me know when you've fixed the issues and update the pull request.

jedwards1211 commented:
Okay, the Big Spring data now parses and stations show up in the correct position on screen.  I still want to write testcases for the trig methods I came up with to derive distance and inclination from the instrument and target heights and whatnot (used in that data as dive depth measurements).  After that, it should finally be ready!

vpicaver commented:
Cool, we can probably port/call that code into cavewhere once we support
dive data. I'm going to try to add dive support with the metacave branch.

Phi|ip

On Tue, Dec 22, 2015 at 10:32 AM, Andy Edwards <notifications@github.com>
wrote:

> Okay, the Big Spring data now parses and stations show up in the correct
> position on screen. I still want to write testcases for the trig methods I
> came up with to derive distance and inclination from the instrument and
> target heights and whatnot (used in that data as dive depth measurements).
> After that, it should finally be ready!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Cavewhere/cavewhere/pull/94#issuecomment-166648211>.
>

jedwards1211 commented:
It's DONE!!!  Merry Christmas!  haha

vpicaver commented:
What a nice present :D

On Thu, Dec 24, 2015 at 9:06 PM, Andy Edwards <notifications@github.com>
wrote:

> It's DONE!!! Merry Christmas! haha
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Cavewhere/cavewhere/pull/94#issuecomment-167178099>.
>

jedwards1211 commented:
I know! :P

jedwards1211 commented:
I'm probably going to take a break from coding outside of work for awhile.  I obsessed over this a bit too much the past week because the end kept seeming closer than it was...

vpicaver commented:
Yea, I know what you mean. It's pretty easy to get burned out.

vpicaver commented:
I'm currently trying to add support for splay shots. One of many things to add support for the metacave format.

vpicaver commented:
I'll test it out, and then probably release 0.08. Just work on Cavewhere
when you want to work on Cavewhere. :D

On Thu, Dec 24, 2015 at 9:09 PM, Andy Edwards <notifications@github.com>
wrote:

> I'm probably going to take a break from coding outside of work for awhile.
> I obsessed over this a bit too much the past week because the end kept
> seeming closer than it was...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Cavewhere/cavewhere/pull/94#issuecomment-167178411>.
>
